DispatchR/Dispatcher.cs

```diff
@@ -94,21 +101,20 @@ public Task RunAsync(CancellationToken token = default)
             }, token);
         }
 
-        protected void Dispatch(List<Dispatchee> dispatchees, CancellationToken token = default)
+        protected static void Dispatch(List<Dispatchee> dispatchees, CancellationToken token = default)
         {
-            lock (_lock)
+            lock (dispatchees)
```

Comment:
Point of the lock was to ensure the for loop wasn't iterating on a collection that could be modified. 
Given that, wouldn't want to lock on `dispatchees` in add/remove methods?